### DISCUSSION TITLE
useSnapshot and re-renders

### DISCUSSION THREAD

**Discussion by nosferatu500** [pmndrs/valtio#279] [Q&A] [answered]
Hey! Thanks again for this great package. I want to clarify one thing.

For manual re-render subscriptions, I use the following construction:
```useShapshot(ProjectStore).project.packageName``` - This value is not used in the current functional component, but I want to re-render this component when the packageName is updated.

Now I am using this construction:
```
// My functional component that I want to update when packageName changes, which is not used in this component.

useSnapshot(ProjectStore).project.packageName;

return (<>something</>)
```
and everything works fine, but I want to clarify the difference between
```useSnapshot(ProjectStore).project.packageName``` and ```useSnapshot(ProjectStore.project).packageName```

There is no difference in my understanding, but I'm not sure.

**Comment by dai-shi** [comment, marked as answer]
`useSnapshot(state).a.b` means subscribing to `state` and rendering if `.a.b` changes.
`useSnapshot(state.a).b` means subscribing to `state.a` and rendering if `.b` changes.
Technically, the latter can be more performant, because it's more granular subscription.
Only the former allows to mutate like `state.a = ...`.

**Reply by nosferatu500** [reply]
Got it! Thanks!
